src/tqec/interop/pyzx/synthesis/strategy.py

```diff
@@ -0,0 +1,40 @@
+"""Collection of strategies for synthesizing a block graph from a ZX graph."""
+
+from collections.abc import Mapping
+from enum import Enum
+from typing import Any
+
+from pyzx.graph.graph_s import GraphS
+
+from tqec.computation.block_graph import BlockGraph
+from tqec.interop.pyzx.positioned import PositionedZX
+from tqec.interop.pyzx.synthesis.positioned import positioned_block_synthesis
+from tqec.utils.position import Position3D
+
+
+class SynthesisStrategy(Enum):
+    """Strategy for synthesizing a block graph from a ZX graph."""
+
+    POSITIONED = "POSITIONED"
+    """Mapping from a :py:class:`~tqec.interop.pyzx.positioned_graph.PositionedZX` instance
+    to a block graph.
+
+    This strategy requires specifying the 3D positions of each vertex explicitly
+    in the ZX graph. Then the conversion converts each vertex by looking at its
+    nearest neighbors to infer the cube kind. The conversion maps each vertex to
+    a cube in the block graph and each edge to a pipe connecting the corresponding
+    cubes.
+    """
+
+
+def block_synthesis(
+    zx_graph: GraphS,
+    strategy: SynthesisStrategy = SynthesisStrategy.POSITIONED,
+    *,
+    positions: Mapping[int, Position3D] | None = None,
+    **_kwargs: dict[str, Any],
+) -> BlockGraph:
+    match strategy:
+        case SynthesisStrategy.POSITIONED:
+            g = PositionedZX(zx_graph, positions or {})
```

Comment:
I would add code here to check that `positions` is not `None` and throw a specific exception if it is instead of letting the constructor of `PositionedZX` throw an exception that will not be descriptive of the real issue.